lib/melodica_inventory/goods/image_operations.ex

```diff
@@ -0,0 +1,50 @@
+defmodule MelodicaInventory.Goods.ImageOperations do
+  @moduledoc false
+
+  import Ecto.Query
+
+  alias MelodicaInventory.Goods.{Item, Image}
+  alias MelodicaInventory.Repo
+
+  def upload_images(item_state, images, opts \\ [])
+  def upload_images(%{item: %Item{}}, nil, opts) do
+    if opts[:required] do
+      {:error, "You need to upload an image"}
+    else
+      {:ok, true}
+    end
+  end
+
+  def upload_images(%{item: %Item{id: item_id}}, uploaded_files, opts) do
+    Enum.map(uploaded_files, &upload_to_cloudex_and_save(&1, item_id))
+    |> format_response
+  end
+
+  def delete_images(images) do
+    if delete_by_public_id(images) == length(images) do
+      Image |> where([i], i.public_id in ^images) |> Repo.delete_all
+    end
+  end
+
+  defp delete_by_public_id(images) do
+    Cloudex.delete(images)
+    |> Enum.reduce(0, fn({:ok, _}, acc) -> 1 + acc end)
```

Comment:
Can we make the delete operation, remove the images from the DB one by one when the `Cloudex.delete` is successful, as in the `upload` operation? This will ensure that if a single delete fails, there won't be zombie records in the DB.